policy/constraints.go

```diff
@@ -90,6 +100,118 @@ func (cc CertConstraint) checkTrustBundles(verifier *cryptoutil.X509Verifier, tr
 	return fmt.Errorf("cert doesn't belong to any root specified by constraint %+q", cc.Roots)
 }
 
+func (cc CertConstraint) checkExtensions(ext []pkix.Extension) error {
+	extensions, err := cc.parseExtensions(ext)
+	if err != nil {
+		return fmt.Errorf("error parsing fulcio cert extensions: %w", err)
+	}
+
+	fields := reflect.VisibleFields(reflect.TypeOf(cc.Extensions))
+	for _, field := range fields {
+		constraintField := reflect.ValueOf(cc.Extensions).FieldByName(field.Name)
+		if constraintField.String() == "" {
+			log.Infof("No constraint for field %s, allowing all values", field.Name)
+			continue
+		}
+		extensionsField := reflect.ValueOf(extensions).FieldByName(field.Name)
+
+		fieldGlob := glob.MustCompile(constraintField.String())
+		if !fieldGlob.Match(extensionsField.String()) {
+			return fmt.Errorf("cert field %s doesn't match constraint %+q", field.Name, constraintField.String())
+		}
+	}
+
+	return nil
+}
+
+// forked from fulcio since it's not exported.
+func (cc CertConstraint) parseExtensions(ext []pkix.Extension) (certificate.Extensions, error) {
```

Comment:
Frustrating that this isn't exported.

Reply:
I'm going to follow up on Sigstore Slack and see if there's a reason they're not exporting [this](https://github.com/sigstore/fulcio/blob/main/pkg/certificate/extensions.go#L341).

Reply:
https://github.com/sigstore/fulcio/pull/1584

Reply:
Ah awesome! That's great news :D will we have to wait for them to cut a release?

Reply:
Nice release has been cut - if we get this bumped up to https://github.com/sigstore/fulcio/releases/tag/v1.4.4 then I reckon this is good to go 😄